target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -289,7 +289,7 @@ pub struct PostUpdatesAtomic<'info> {
     #[account(mut, seeds = [TREASURY_SEED.as_ref()], bump)]
     /// CHECK: This is just a PDA controlled by the program. There is currently no way to withdraw funds from it.
     pub treasury:             AccountInfo<'info>,
-    #[account(init, payer = payer, space = PriceUpdateV1::LEN)]
+    #[account(init_if_needed, constraint = price_update_account.write_authority == Pubkey::default() || price_update_account.write_authority == payer.key(), payer = payer, space = PriceUpdateV1::LEN)]
```

Comment:
It can only be equal to Pubkey::default() (i.e. all zeros) when it just got initialized.

Reply:
maybe worth leaving a code comment about ^

Reply:
Done